spec/intro.dd

```diff
@@ -67,6 +67,43 @@ $(OL
 )
 
 
+$(H2 Memory Model)
+
+    $(P A $(I byte) is the fundamental unit of storage. Each byte has 8 bits. Each byte has
```

Comment:
This would exclude certain platforms.

Reply:
such micro controllers with e.g. bit-banding.

Reply:
We don't care about non-8 bit platforms. That doesn't actually precluded writing a D compiler specifically for such platforms, they just wouldn't be compliant.

For example, C supports platforms with 32 bit bytes. However, I can't think of any non-trivial C program that wouldn't need to be completely rewritten to work on such a machine. So it is kinda pointless to pretend to support them.

Reply:
cc @JinShil how much have you done with ARM cortex m3/m4?

Reply:
The ARM Cortex-M (a 32-bit microcontroller) has a feature called bit-banding where a 32-bit address addresses a single bit in memory.  It works for both SRAM and memory-mapped IO.  I rely on it to reduce code size and increase performance in my memory-mapped IO library:  

https://github.com/JinShil/stm32f42_discovery_demo/blob/master/source/stm32f42/mmio.d#L132-L142

I currently encapsulate using D's metaprogramming features:

Reply:
I'd never heard of bit banding before, thanks for the links. I think it is a special case for ARM and outside the purview of the spec. Being outside of the spec, of course, doesn't prevent writing D code to work with bit banding.